### PR TITLE
do not send invoice emails to projects with credits autogenerated and balance zero

### PR DESCRIPTION
**Pull request by nickpell**
http://manage.dimagi.com/default.asp?167095#930566

@benrudolph @orangejenny 

should go out with today's deploy

### CODE REVIEW

**Comment by benrudolph** [comment]
:+1: should i wait for tests or is this urgent?

**Comment by nickpell** [comment]
@emord is holding off deploy until later, so can wait on tests.

File: corehq/apps/accounting/invoicing.py [hunk 1]
```diff
@@ -199,7 +199,10 @@ def create_invoice_for_subscription(self, subscription):
 
         record = BillingRecord.generate_record(invoice)
         try:
-            record.send_email()
+            if subscription.auto_generate_credits and not invoice.balance:
+                record.skipped_email = True
```
Comment (orangejenny) [inline corehq/apps/accounting/invoicing.py]: what does this do? it doesn't look like `record` gets saved...?

Comment (nickpell) [inline corehq/apps/accounting/invoicing.py]: good catch

Comment (benrudolph) [inline corehq/apps/accounting/invoicing.py]: :+1:

**Pull request merged**